src/c/c_client.cpp

```diff
@@ -30,517 +30,654 @@
+void get_script(void *c_client,
+                const char *key,
+		const size_t key_length,
+                const char **script,
+		size_t *script_length)
 {
-  /* Get the script stored in the database
-  */
-  Client* s = (Client *)c_client;
-  std::string key_str = std::string(key, key_length);
-  std::string_view script_str_view = s->get_script(key_str);
+  // Sanity check params
+  if (NULL == c_client || NULL == key || NULL == script || NULL == script_length)
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+  
+  Client *s = reinterpret_cast<Client *>(c_client);
+  std::string key_str(key, key_length);
+  std::string_view script_str_view(s->get_script(key_str));
+
   (*script) = script_str_view.data();
   (*script_length) = script_str_view.size();
-  return;
 }
 
+// Run  a script function in the database
 extern "C"
-void run_script(void* c_client,
-                const char* key,
+void run_script(void *c_client,
+                const char *key,
                 const size_t key_length,
-                const char* function,
+                const char *function,
                 const size_t function_length,
-                const char** inputs,
-                const size_t* input_lengths,
+                const char **inputs,
+                const size_t *input_lengths,
                 const size_t n_inputs,
-                const char** outputs,
-                const size_t* output_lengths,
+                const char **outputs,
+                const size_t *output_lengths,
                 const size_t n_outputs)
 {
-  /* This function runs a script function in the database
-  */
-  Client* s = (Client *)c_client;
-  std::string key_str = std::string(key, key_length);
-  std::string function_str = std::string(function, function_length);
+  // Sanity check params
+  if (NULL == c_client || NULL == key || NULL == function || NULL == inputs ||
+      NULL == input_lengths || NULL == outputs || NULL == output_lengths) {
+    return; // Nothing we can do, so just bail. (error reporting to come later)
+  }
+
+  std::string key_str(key, key_length);
+  std::string function_str(function, function_length);
 
   std::vector<std::string> input_vec;
-  for(size_t i=0; i<n_inputs; i++)
-    input_vec.push_back(std::string(inputs[i], input_lengths[i]));
+  if (1 != n_inputs || 0 != input_lengths[0]) {
+    for (size_t i=0; i<n_inputs; i++) {
+      input_vec.push_back((NULL != inputs[i] && 0 != input_lengths[i]) ?
+			  std::string(inputs[i], input_lengths[i]) :
+			  std::string(""));
+    }
+  }
 
   std::vector<std::string> output_vec;
-  for(size_t i=0; i<n_outputs; i++)
-    output_vec.push_back(std::string(outputs[i], output_lengths[i]));
+  if (1 != n_outputs || 0 != output_lengths[0]) {
+    for (size_t i=0; i<n_outputs; i++) {
+      output_vec.push_back((NULL != outputs[i] && 0 != output_lengths[i]) ?
+			   std::string(outputs[i], output_lengths[i]) :
```

Comment:
This seems to implicitly allow for output tensor names that are empty.  is that ok?  It seems like that's something we shouldn't allow.

Reply:
See previous